### PR TITLE
Bug fix when creating dataset from zarr

### PR DESCRIPTION
**Pull request by b8raoult**
Addressing https://github.com/ecmwf/anemoi-datasets/issues/46

### CODE REVIEW

File: src/anemoi/datasets/create/__init__.py [hunk 1]
```diff
@@ -487,6 +487,7 @@ def _run(self):
         assert chunks == self.dataset.get_zarr_chunks(), (chunks, self.dataset.get_zarr_chunks())
 
         def sanity_check_config(a, b):
+            return
```
Comment (mchantry) [inline src/anemoi/datasets/create/__init__.py:490]: Why return here? Rest of code won't be used.
Should this be on line 498?

File: src/anemoi/datasets/create/input.py [hunk 2]
```diff
@@ -108,14 +108,15 @@ def _data_request(data):
     for field in data:
         try:
             if date is None:
-                date = field.datetime()["valid_time"]
+                date = field.metadata("valid_datetime")
 
-            if field.datetime()["valid_time"] != date:
+            if field.metadata("valid_datetime") != date:
```
Comment (mchantry) [inline src/anemoi/datasets/create/input.py:113]: What is this line doing?

Comment (mchantry) [reply src/anemoi/datasets/create/input.py:113]: i.e. why continue?

File: src/anemoi/datasets/create/__init__.py (hunk 1 above)
Comment (b8raoult) [reply src/anemoi/datasets/create/__init__.py:490]: The message below has confused a few users. So, I have temporarily disabled it.

File: src/anemoi/datasets/create/input.py (hunk 2 above)
Comment (b8raoult) [reply src/anemoi/datasets/create/input.py:113]: We only care about one date. So we consider only the fields with the same valid date as the first.

**Review by floriankrb — APPROVED** [review]

**Pull request merged**